test/Command.js

```diff
@@ -187,7 +184,83 @@ describe("Command", () => {
   });
 
   describe(".runPreparations()", () => {
-    it("needs tests");
+    describe(".filteredPackages", () => {
+      let testDir;
+
+      beforeEach(() => initFixture("UpdatedCommand/basic").then((dir) => {
+        testDir = dir;
+      }));
+
+      class TestCommand extends Command {}
+
+      function cli(cmd, ...args) {
+        return execa.sync(cmd, args, { cwd: testDir }).stdout.toString("utf-8");
+      }
+
+      function run(opts) {
+        const cmd = new TestCommand([], opts, testDir);
+        cmd.run();
+        return cmd;
+      }
+
+      it("--scope should filter packages", () => {
+        const { filteredPackages } = run({ scope: ["package-2", "package-4"] });
+        expect(filteredPackages.length).toEqual(2);
+        expect(filteredPackages[0].name).toEqual("package-2");
+        expect(filteredPackages[1].name).toEqual("package-4");
+      });
+
+      it("--since should return all packages if no tag is found", () => {
+        const { filteredPackages } = run({ since: "" });
+        expect(filteredPackages.length).toEqual(5);
+      });
+
+      it("--since should return packages updated since the last tag", () => {
+        cli("git", "tag", "1.0.0");
+        cli("touch", "packages/package-2/random-file");
+        cli("git", "add", ".");
+        cli("git", "commit", "-m", "test");
+
+        const { filteredPackages } = run({ since: "" });
+        expect(filteredPackages.length).toEqual(2);
+        expect(filteredPackages[0].name).toEqual("package-2");
+        expect(filteredPackages[1].name).toEqual("package-3");
+      });
+
+      it("--since \"ref\" should return packages updated since the specified ref", () => {
+        // We first tag, then modify master to ensure that specifying --since will override checking against
```

Comment:
This was the addition that caught the bug where the latest tag would override the specified `ref` in `--since`. I annotated the spot above that fixes it.